src/components/TextBox.tsx

```diff
@@ -52,14 +55,16 @@ export default function TextBox({ field }: TextBoxProps): JSX.Element {
 	function onTextBoxInputChange(event: React.ChangeEvent<HTMLInputElement>): void {
 		let userInputValue = event.target.value;
 
-		// Only trigger debounced from local value change if the
-		// user defined input converts to a number
-		if (!Number.isNaN(userInputValue)) {
-			setIsDebouncing(true);
+		// Only set local input value if the user defined input is a valid input
+		if (isValidInput(userInputValue)) {
+			if (userInputValue !== '.') {
+				// Only trigger debounce if the input is not a solo decimal
```

Comment:
You could avoid this check here if `isValidInput()` returns `false` when it's `'.'`